packages/engine/Source/DataSources/ModelVisualizer.js

```diff
@@ -444,8 +446,10 @@ ModelVisualizer.prototype.getBoundingSphere = function (entity, result) {
 
   const scene = this._scene;
   const globe = scene.globe;
-  const ellipsoid = globe.ellipsoid;
-  const terrainProvider = globe.terrainProvider;
+  // Check if the globe is undefined and use a substitute ellipsoid if it is not
+  const ellipsoid = globe ? globe.ellipsoid : Ellipsoid.WGS84;
```

Comment:
Instead of assuming the ellipsoid value, I think it would actually be better to return early and skip some of these unneeded calculations below when `globe` is undefined.

I believe when the globe is undefined, we can simply  skip the `hasHeightReference` block entirely as there is no ellipsoid to offset the position against, and we can just use the position as-is.

Reply:
thanks for the feedback! I added a condition to `hasHeightReference` that skips that block if the globe is undefined. I also removed the ellipsoid being instantiated as `Ellipsoid.WGS84` if one isn't present, and moved the ellipsoid being defined to within the `hasHeightReference` block as you would need a globe for this anyway